site/content/docs/howto/supervisor/configure-supervisor-with-github.md

```diff
@@ -0,0 +1,177 @@
+
+"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
+
+"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
+
+[## Configure the Supervisor
+
+Create a [GitHubIdentityProvider](https://github.com/vmware-tanzu/pinniped/blob/main/generated/{{< latestcodegenversion >}}/README.adoc#githubidentityprovider) in the same namespace as the Supervisor.
+
+The simplest example uses https://github.com as the source of identity, and will allow any user to log in.
+Note that you do not need to explicitly specify a GitHub host since `github.com` is the default.
+
+```yaml
+apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+kind: GitHubIdentityProvider
+metadata:
+  namespace: pinniped-supervisor
+  name: github-dot-com
+spec:
+  client:
+    secretName: github-dot-com-client-credentials
+  allowAuthentication:
+    organizations:
+      policy: AllGitHubUsers
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: pinniped-supervisor
+  name: github-dot-com-client-credentials
+type: secrets.pinniped.dev/github-client
+stringData:
+
+  # The "Client ID" from the GitHub App or GitHub OAuth App.
+  clientID: "<your-client-id>"
+
+  # The "Client secret" from the GitHub App or GitHub OAuth App.
+  clientSecret: "<your-client-secret>"
+```
+
+For another example, let's fill out all fields.
+Here we will configure a host and certificate for a GitHub Enterprise Server installation.
+
+```yaml
+apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+kind: GitHubIdentityProvider
+metadata:
+  namespace: pinniped-supervisor
+  name: github-enterprise
+spec:
+  githubAPI:
+    
+    # Only the hostname or IP address and optional port, without the protocol.
+    # Pinniped will always use HTTPS.
+    host: github.enterprise.tld
+    
+    tls:
+      
+      # Must be a base64-encoded PEM cert ("-----BEGIN CERTIFICATE-----") or bundle
+      certificateAuthorityData: LS0tLS1CRUdJTiBDRVJUSUZJQ0FU....
+
+  client:
+    secretName: github-enterprise-client-credentials
+  allowAuthentication:
+
+    # "policy: OnlyUsersFromAllowedOrganizations" restricts authentication to only those users who belong in at least one of the "allowed" organizations.
```

Comment:
Lets force a word wrap here, probably at "to only", it doesn't render as nice with long lines.

Reply:
done